analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TypeHelperTest.cs

```diff
@@ -452,6 +452,43 @@ class Test<T> {{typeConstraint}}
             fieldSymbol.Type.IsNullableValueType().Should().BeFalse();
         }
 
+        [DataTestMethod]
+        [DataRow("", "AttributeTargets")]
+        [DataRow("where T: Enum", "T")]
+        [DataRow("where T: struct, Enum", "T")]
+        [DataRow("where T: Enum", "T?")] // Here ? means either nullable value type or nullable reference type (unbound generic) and T? is an Enum
```

Comment:
It took me some time to understand this. The key point I was missing is that we're in "nullable context". Can you embedded that note in the comment?